lib/LaTeXML/Package/LaTeX.pool.ltxml

```diff
@@ -363,8 +363,9 @@ DefConstructorI(T_CS('\end{document}'), undef, sub {
       && (!($ifstack = $STATE->lookupValue('if_stack')) || !$$ifstack[0])) { }    # OK!
     else {
       my @lines = ();
-      while (!$STATE->isValueBound('current_environment', 0)
-        || ($STATE->valueInFrame('current_environment', 0) ne 'document')) {
+      while ((!$STATE->isValueBound('current_environment', 0)
+          || ($STATE->valueInFrame('current_environment', 0) ne 'document'))
+        && ($STATE->getFrameDepth > 0)) {
```

Comment:
Ah, I see you've added the guard, scratch my last comment and sorry for the noise.